qcodes/dataset/descriptions/detect_shapes.py

```diff
@@ -55,7 +55,10 @@ def detect_shape_of_measurement(
     shapes: Dict[str, Tuple[int, ...]] = {}
 
     for param_name in array_shapes.keys():
-        shapes[param_name] = tuple(loop_shape) + array_shapes[param_name]
+        total_shape = tuple(loop_shape) + array_shapes[param_name]
+        if total_shape == ():
+            total_shape = (1,)
```

Comment:
so this means we had a bug before that scalar parameters would get `()` as opposed to `(1,)` - do i understand this correctly? if so, i wonder if the line `array_shapes[param.full_name] = ()` above is correct and should be changed to `... = (1,)`? or i'm totally misunderstanding the why here.

Reply:
I don't think so. This is really only cover the case where you measure a 0d experiment with a scalar parameter so that both `loop_shape` and `array_shapes[param_name]` are empty. If you set array_shapes to (1,) you would expand 1d sweeps of scalar parameters from (n,) to (n,1) which I don't think we want

Reply:
ah, 0d, of course.